rust/connlib/tunnel/src/client.rs

```diff
@@ -58,6 +58,31 @@ impl<CB> ClientTunnel<CB>
 where
     CB: Callbacks + 'static,
 {
+    pub fn set_resources(
```

Comment:
This has quite some edge-cases that I think are covered but I will add some unit tests before opening the PR for review

Reply:
unit testing the `Tunnel` is still a bit hard.